### DISCUSSION TITLE
[FEATURE] Extendibility of AiServices

### DISCUSSION THREAD

**Discussion by er3n** [langchain4j/langchain4j#202] [Ideas]
Hi, I am looking for ways to extend AIServices. The main reason behind this, for instance, is illustrated in the sample below. It would be great to read system messages from a config file or other resources.

Unfortunately, the current AIServices class is not extendable for me. To implement the feature described below, I need to clone AIServices, which is not maintainable.

```
public interface Assistant {
    @SystemMessageFromEnv(value="myApp.systemMessage") 
    Answer ask(Question question);
}
```

**Comment by chenhaoqiang** [comment]
This can be implemented like this, using annotation @V, then the parameter 'systemMessage' can be  read from a config file or other resources. Hope this helps you.

```
public interface Assistant {
    @SystemMessage("{{systemMessage}}") 
    Answer ask(Question question, @V("systemMessage") String systemMessage);
}
```

**Reply by er3n** [reply]
Thank you, I will go with this solution
